src/traces/ohlc/helpers.js

```diff
@@ -95,14 +95,65 @@ exports.makeTransform = function(traceIn, state, direction) {
 };
 
 exports.getFilterFn = function(direction) {
+    return new _getFilterFn(direction);
+};
+
+function _getFilterFn(direction) {
+    var isPrevThisDirection = null;
+    var cPrev = null;
+    var fn;
+
     switch(direction) {
         case 'increasing':
-            return function(o, c) { return o <= c; };
+            fn = function(o, c) {
+                if(o === c) {
```

Comment:
checked all to ensure they're numeric in https://github.com/plotly/plotly.js/pull/1655/commits/597adf96181b40a4eabc7b3622e919d19e1864a0